### PR TITLE
Add hot reload support

### LINKED ISSUE #366: Add Hot-Reload support
TUM-Live Server should serve the current GOHTML pages when they are changed. 
Currently a server restart is needed.

### PR DESCRIPTION
**Pull request by mono424**
- Hot reload of .gohtml and static files.
- Added tailwind jit to `npm run build-dev` to support updated tailwind CSS files on .gohtml change.

https://user-images.githubusercontent.com/17506411/166139202-858e3990-c8aa-4e17-8810-f850fecc2aa9.mov

### CODE REVIEW

File: tools/template_executor.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package tools
```
Comment (joschahenningsen) [inline tools/template_executor.go:1]: I think this could be in the web package or a sub package of web, or is it used somewhere else?

**Review by joschahenningsen — APPROVED** [review]
Code looks good and works flawless, amazing! Just one thought that is not important though :)

**Comment by joschahenningsen** [comment]
If you merge dev, the staticcheck passes again :)

File: tools/template_executor.go (hunk 1 above)
Comment (mono424) [reply tools/template_executor.go:1]: In `web` and in `tools.Middlewares` should I still move it, I would be okay with both actually as it is much more used in web than in tools 👍

**Comment by mono424** [comment]
@joschahenningsen I merged dev. Dev & this branch not running locally because of `error="open fullchain.pem: no such file or directory"`. Should work tho, I think its my local env. :)

File: tools/template_executor.go (hunk 1 above)
Comment (joschahenningsen) [reply tools/template_executor.go:1]: You're right :+1:

**Comment by joschahenningsen** [comment]
@mono424 I merged again and forgot to comment out the saml config, it should run again now :)

**Pull request merged**